Fix Turbo v2 setup issues when following official Midnight Network tutorial

ShawnTseng opened the pull request:
While following the official Midnight Network tutorial for building the example-bboard DApp (https://docs.midnight.network/develop/tutorial/creating/bboard-dapp#compiling-and-running-the-dapp), several issues prevented `npx turbo build` from running correctly. 

This PR fixes those issues:

- **Missing `packageManager` field**: Added to root package.json as required by Turbo v2.
- **`turbo.json` location**: Moved to the repository root to comply with Turbo v2 expectations.
- **Deprecated `outputMode`**: Replaced with `outputLogs` in turbo.json.
- **`pipeline` field**: Renamed to `tasks` in turbo.json for Turbo v2.
- **`dependsOn` syntax**: Updated to correctly reference `compact` tasks.
- **Turbo cache**: Added `.turbo` to .gitignore.
- **Standalone script**: Runs locally using ts-node/esm on the original .ts files, after pulling required Docker images.

These fixes make the build scripts run successfully while following the tutorial instructions.

ShawnTseng committed:
fix(root): add packageManager field in package.json to comply with Turbo v2 requirement

ShawnTseng committed:
fix(root): move turbo.json to repo root to comply with Turbo v2 requirement

ShawnTseng committed:
fix(turbo): replace deprecated `outputMode` with `outputLogs` in turbo.json for Turbo v2

ShawnTseng committed:
fix(turbo): rename `pipeline` to `tasks` in turbo.json for Turbo v2

ShawnTseng committed:
fix(turbo): update `dependsOn` syntax for Turbo v2 to reference correct compact task

ShawnTseng committed:
chore: add .turbo to .gitignore

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/midnightntwrk/example-bboard?pullRequest=1002) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/midnightntwrk/example-bboard?pullRequest=1002) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/midnightntwrk/example-bboard?pullRequest=1002) it.</sub>

claudebarde commented on package.json:
```diff
@@ -13,6 +13,7 @@
       "contract"
     ]
   },
+  "packageManager": "npm@10.2.3",
```
I am reluctant to force our developers to use a specific version of NPM, above all because this one is not the most recent one (11.5.2 is).
Can you explain this choice or update your PR accordingly?
Thank you.

gilescope replied:
Specifying a version is normal here. But agreed no reason not to use the latest version

gilescope committed:
Merge branch 'main' into main

ShawnTseng committed:
Fix: update packageManager to npm@11.5.2

ShawnTseng committed:
config: Set packageManager to yarn@1.22.22

ShawnTseng committed:
chore: run standalone using ts-node/esm from source instead of compiled JS

ShawnTseng committed:
fix: run standalone fully via Docker to avoid Node ESM/CJS conflict

ShawnTseng committed:
Specify Node 20+ requirement in package.json

- Added "engines" field to enforce Node version >=20
- Standalone script remains using ts-node/esm with experimental specifier resolution

ShawnTseng committed:
fix: disable package manager check in turbo.json

Add dangerouslyDisablePackageManagerCheck flag to resolve build failures
when running npx turbo build. This allows flexibility in package manager
usage without requiring a specific npm version in packageManager field.

ref: https://turborepo.com/docs/reference/configuration#dangerouslydisablepackagemanagercheck

ShawnTseng committed:
chore: remove unnecessary whitespace

bytrangle reviewed (commented):
Look neat to me. There are some changes in Turbo that are not documented unless you dig deep into their blog. These changes look neat to me.
